pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentRestletResource.java

```diff
@@ -678,6 +699,81 @@ public SuccessResponse reloadAllSegments(
     return new SuccessResponse(JsonUtils.objectToString(perTableMsgData));
   }
 
+  @POST
+  @Path("segments/{tableName}/batchReload")
+  @Authorize(targetType = TargetType.TABLE, paramName = "tableName", action = Actions.Table.RELOAD_SEGMENT)
+  @Authenticate(AccessType.UPDATE)
+  @Produces(MediaType.APPLICATION_JSON)
+  @ApiOperation(value = "Reload with a map from instances to segments for more flexible control", notes = "Reload "
+      + "with a map from instances to segments for more flexible control")
+  public SuccessResponse reloadSegments(
+      @ApiParam(value = "Name of the table", required = true) @PathParam("tableName") String tableName,
+      @ApiParam(value = "OFFLINE|REALTIME") @QueryParam("type") String tableTypeStr,
+      @ApiParam(value = "Whether to force server to download segment") @QueryParam("forceDownload")
+      @DefaultValue("false") boolean forceDownload,
+      @ApiParam(value = "Map from instances to segments to reload", required = true) @QueryParam("reloadMap")
```

Comment:
Suggest renaming it to `instanceToSegmentsMap` to be more specific

Reply:
Consider also allowing reloading a list of segments by adding a `segments` query parameter

Reply:
The segments should be provided by that Map as well, ie which segments to reload on which instances.